packages/discovery-react-components/src/components/SearchFacets/utils/searchFacetInterfaces.ts

```diff
@@ -33,3 +41,25 @@ export interface CollectionItem {
   id: string;
   label: string;
 }
+
+export interface QueryAggregationWithName extends DiscoveryV2.QueryAggregation {
+  field?: string;
```

Comment:
why is `field` optional?

Reply:
The issue I keep running into here is that `DiscoveryV2.QueryAggregation` doesn't have `field` required (although `DiscoveryV2.QueryTermAggregation`, etc., which inherit from it do), but it's the type that is returning in the `QueryResponse` in the search client in `DiscoverySearch`. Therefore, there are errors in `DiscoverySearch` if it's not optional here between getting the response and passing it off to `buildAggregationQuery`. I haven't come up with a better solution here so far, but I'll keep thinking...

Reply:
are you able to use a type guard to enforce it to be a `QueryTermAggregation` instead of `QueryAggregation`?